src/bin/firebase.ts

```diff
@@ -11,160 +12,11 @@ if (!semver.satisfies(nodeVersion, pkg.engines.node)) {
-logger.debug("Time:         ", new Date().toString());
-if (utils.envOverrides.length) {
-  logger.debug("Env Overrides:", utils.envOverrides.join(", "));
-}
-logger.debug("-".repeat(70));
-logger.debug();
-
-import { enableExperimentsFromCliEnvVariable } from "../experiments";
-import { fetchMOTD } from "../fetchMOTD";
-
-enableExperimentsFromCliEnvVariable();
-fetchMOTD();
-
-process.on("exit", (code) => {
-  code = process.exitCode || code;
-  if (!process.env.DEBUG && code < 2 && fsutils.fileExistsSync(logFilename)) {
-    fs.unlinkSync(logFilename);
-  }
-
-  if (code > 0 && process.stdout.isTTY) {
-    const lastError = configstore.get("lastError") || 0;
-    const timestamp = Date.now();
-    if (lastError > timestamp - 120000) {
-      let help;
-      if (code === 1 && cmd) {
-        help = "Having trouble? Try " + clc.bold("firebase [command] --help");
-      } else {
-        help = "Having trouble? Try again or contact support with contents of firebase-debug.log";
-      }
-
-      if (cmd) {
-        console.log();
-        console.log(help);
-      }
-    }
-    configstore.set("lastError", timestamp);
-  } else {
-    configstore.delete("lastError");
-  }
-
-  // Notify about updates right before process exit.
-  try {
-    const installMethod = !process.env.FIREPIT_VERSION ? "npm" : "automatic script";
-    const updateCommand = !process.env.FIREPIT_VERSION
-      ? "npm install -g firebase-tools"
-      : "curl -sL https://firebase.tools | upgrade=true bash";
-
-    const updateMessage =
-      `Update available ${clc.gray("{currentVersion}")} → ${clc.green("{latestVersion}")}\n` +
-      `To update to the latest version using ${installMethod}, run\n${clc.cyan(updateCommand)}\n` +
-      `For other CLI management options, visit the ${marked(
-        "[CLI documentation](https://firebase.google.com/docs/cli#update-cli)",
-      )}`;
-    // `defer: true` would interfere with commands that perform tasks (emulators etc.)
-    // before exit since it installs a SIGINT handler that immediately exits. See:
-    // https://github.com/firebase/firebase-tools/issues/4981
-    updateNotifier.notify({ defer: false, isGlobal: true, message: updateMessage });
-  } catch (err) {
-    // This is not a fatal error -- let's debug log, swallow, and exit cleanly.
-    logger.debug("Error when notifying about new CLI updates:");
-    if (err instanceof Error) {
-      logger.debug(err);
-    } else {
-      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-      logger.debug(`${err}`);
-    }
-  }
-});
-
-process.on("uncaughtException", (err) => {
-  errorOut(err);
-});
-
-if (!handlePreviewToggles(args)) {
-  // determine if there are any arguments. if not, display help
-  if (!args.length) {
-    client.cli.help();
-  } else {
-    cmd = client.cli.parse(process.argv);
-  }
+// we short-circuit the normal process for MCP
+if (isEnabled("mcp") && process.argv[2] === "experimental:mcp") {
```

Comment:
Probably OK to just make the command `firebase mcp`, since its already behind an experiment flag

Reply:
So I actually intend to remove the experiment flag before we launch - most MCP servers are installed and run with an `npx` command so I want to make sure this can be run on a fresh machine's install without needing to have done any pre-work (I have additional work to do around not requiring auth etc).

So the idea is someone could just add `npx -y firebase-tools experimental:mcp` as their MCP server command and even if they've never used Firebase before it gets them started. If we require an experiment flag, they'd have to download the CLI separately.

I hoped that there was a command line flag to enable an experiment but don't see one so this is my current approach. Maybe I should add that flag instead.